storage/src/main/scala/ch/epfl/bluebrain/nexus/storage/Storages.scala

```diff
@@ -140,9 +141,13 @@ object Storages {
   }
 
   object PathExistence {
-    final case object PathExists       extends PathExistence
-    final case object PathDoesNotExist extends PathExistence
-    type PathExists       = PathExists.type
+    final case object PathExists       extends PathExistence {
+      val exists = true
+    }
+    final case object PathDoesNotExist extends PathExistence {
+      val exists = false
+    }
+    type PathExists = PathExists.type
```

Comment:
I didn't realise you can do this, I was always using traits to achieve the same thing. Nice